wrappers/nodejs/src/addon.cpp

```diff
@@ -2168,46 +2490,22 @@ class DevicesChangedCallbackInfo : public MainThreadCallbackInfo {
   DevicesChangedCallbackInfo(rs2_device_list* r,
       rs2_device_list* a, RSContext* ctx) :
       removed_(r), added_(a), ctx_(ctx) {}
-  virtual ~DevicesChangedCallbackInfo() {
-    rs2_delete_device_list(removed_);
-    rs2_delete_device_list(added_);
-  }
+  virtual ~DevicesChangedCallbackInfo() {}
   virtual void Run() {
-    v8::Local<v8::Array> removed_array = Nan::New<v8::Array>();
-    v8::Local<v8::Array> added_array = Nan::New<v8::Array>();
-    uint32_t removed_index = 0;
-    uint32_t added_index = 0;
-
-    // prepare removed devices
-    if (ctx_->devices.size()) {
-      for (auto it = ctx_->devices.begin(); it != ctx_->devices.end(); ++it) {
-        RSDevice* rsdevice =
-            Nan::ObjectWrap::Unwrap<RSDevice>(Nan::New(*(*it)));
-        if (rs2_device_list_contains(removed_, rsdevice->dev, &ctx_->error)) {
-          removed_array->Set(removed_index++, Nan::New(*it));
-        }
-      }
-    }
-
-    // // TODO: Do we need to remove all removed deivces from the devices list?.
-    // devices.remove_if([&removed](v8::Persistent<v8::Object> jsdev) {
-    //   RSDevice* rsdevice =
-    //       Nan::ObjectWrap::Unwrap<RSDevice>(Nan::New(jsdev));
-    //   if (rs2_device_list_contains(removed, rsdevice->dev))
-    //     return true;
-    //   else
-    //     return false;
-    // });
-
-    // prepare newly connected devices
-    auto cnt = rs2_get_device_count(added_, &ctx_->error);
-    for (int32_t i=0; i < cnt; i++) {
-      auto jsdev = ctx_->CreateNewDevice(ctx_->device_list, i, &ctx_->error);
-      added_array->Set(added_index++, jsdev);
-    }
-    v8::Local<v8::Value> args[2] = {removed_array, added_array};
+    Nan::HandleScope scope;
+    v8::Local<v8::Value> rmlist;
+    v8::Local<v8::Value> addlist;
+    if (removed_)
+      rmlist = RSDeviceList::NewInstance(removed_);
+    else
+      rmlist = Nan::Undefined();
+    if (added_)
+      addlist = RSDeviceList::NewInstance(added_);
+    else
+      addlist = Nan::Undefined();
```

Comment:
use `? :` please

Reply:
If we use ? :, there will be compile error like: "operands to ?: have different types ‘v8::Local<v8::Object>’ and ‘v8::Local<v8::Primitive>" as they are of different type.

Reply:
Use `static_cast` to the type of `addlist`?

Reply:
I tried it, it compiles, but that code will be:
```
addlist = added_ ? static_cast<v8::Local<v8::Value>>(RSDeviceList::NewINstance(added_)) : static_cast<v8::Local<v8::Value>>(Nan::Undefined())
```
it seems that is too long, maybe we can keep to divide them ?